segfault on library container that doesn't exist

## Version of Singularity:

3.2.1-660.g4c8a84050

### Expected behavior
```
$ singularity exec library://no-container true
FATAL:   Unable to handle library://no-container uri: failed to find library://no-container: Error container does not exist.
I'm sorry Dave, I'm afraid I can't do that...
```
(Or something along those lines.)
### Actual behavior

```
$ singularity exec library://no-container true
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xb0 pc=0xff9fd9]

goroutine 1 [running]:
github.com/sylabs/singularity/cmd/internal/cli.handleLibrary(0x7ffe4d2e03a1, 0x16, 0xc0003a0140, 0x19, 0x7ffe4d2e03a9, 0xe, 0xc3, 0x30)
	cmd/internal/cli/actions.go:149 +0x2f9
github.com/sylabs/singularity/cmd/internal/cli.replaceURIWithImage(0x1c18ca0, 0xc00000eda0, 0x2, 0x2)
	cmd/internal/cli/actions.go:227 +0x3ff
github.com/sylabs/singularity/cmd/internal/cli.actionPreRun(0x1c18ca0, 0xc00000eda0, 0x2, 0x2)
	cmd/internal/cli/actions.go:43 +0x125
github.com/spf13/cobra.(*Command).execute(0x1c18ca0, 0xc0000320e0, 0x2, 0x2, 0x1c18ca0, 0xc0000320e0)
	vendor/github.com/spf13/cobra/command.go:755 +0x4e2
github.com/spf13/cobra.(*Command).ExecuteC(0x1c21fe0, 0x4, 0xc0004f4240, 0x27)
	vendor/github.com/spf13/cobra/command.go:850 +0x2bb
github.com/sylabs/singularity/cmd/internal/cli.ExecuteSingularity()
	cmd/internal/cli/singularity.go:246 +0x13c
main.main()
	cmd/singularity/cli.go:17 +0x20
```
ewww
### Steps to reproduce behavior

```
$ singularity exec library://no-container true
```

--- a/cmd/internal/cli/actions.go
+++ b/cmd/internal/cli/actions.go
@@ -140,9 +140,16 @@ func handleLibrary(u, libraryURL string) (string, error) {
 
 	imageRef := libraryhelper.NormalizeLibraryRef(u)
 
-	libraryImage, _, err := c.GetImage(ctx, imageRef)
+	libraryImage, existOk, err := c.GetImage(ctx, imageRef)
 	if err != nil {
 		return "", err
+	}
+	if !existOk {
+		return "", fmt.Errorf("image does not exist in the library: %s", imageRef)
+	}
+	if libraryImage == nil {
+		// It should never get here
+		return "", fmt.Errorf("failed getting image from the library: %s", imageRef)
 	}
 
 	imageName := uri.GetName("library://" + imageRef)

--- a/cmd/internal/cli/actions.go
+++ b/cmd/internal/cli/actions.go
@@ -147,10 +147,6 @@ func handleLibrary(u, libraryURL string) (string, error) {
 	if !existOk {
 		return "", fmt.Errorf("image does not exist in the library: %s", imageRef)
 	}
-	if libraryImage == nil {
-		// It should never get here
-		return "", fmt.Errorf("failed getting image from the library: %s", imageRef)
-	}
 
 	imageName := uri.GetName("library://" + imageRef)
 	imagePath := cache.LibraryImage(libraryImage.Hash, imageName)

--- a/internal/app/singularity/pull.go
+++ b/internal/app/singularity/pull.go
@@ -65,10 +65,6 @@ func LibraryPull(name, ref, transport, fullURI, libraryURI, keyServerURL, authTo
 	if !existOk {
 		return fmt.Errorf("image does not exist in the library: %s", imageRef)
 	}
-	if libraryImage == nil {
-		// It should never get here
-		return fmt.Errorf("failed getting image from the library: %s", imageRef)
-	}
 
 	imagePath := cache.LibraryImage(libraryImage.Hash, imageName)
 	exists, err := cache.LibraryImageExists(libraryImage.Hash, imageName)

--- a/internal/app/singularity/pull.go
+++ b/internal/app/singularity/pull.go
@@ -63,10 +63,11 @@ func LibraryPull(name, ref, transport, fullURI, libraryURI, keyServerURL, authTo
 		return fmt.Errorf("while getting image info: %v", err)
 	}
 	if !existOk {
-		return fmt.Errorf("image does not exist in the library")
+		return fmt.Errorf("image does not exist in the library: %s", imageRef)
 	}
 	if libraryImage == nil {
-		return fmt.Errorf("failed getting image from the library")
+		// It should never get here
+		return fmt.Errorf("failed getting image from the library: %s", imageRef)
 	}
 
 	imagePath := cache.LibraryImage(libraryImage.Hash, imageName)

--- a/internal/pkg/build/sources/conveyorPacker_library.go
+++ b/internal/pkg/build/sources/conveyorPacker_library.go
@@ -65,10 +65,6 @@ func (cp *LibraryConveyorPacker) Get(b *types.Bundle) (err error) {
 	if !existOk {
 		return fmt.Errorf("image does not exist in the library: %s", imageRef)
 	}
-	if libraryImage == nil {
-		// It should never get here
-		return fmt.Errorf("failed getting image from the library: %s", imageRef)
-	}
 
 	imageName := uri.GetName("library://" + imageRef)
 	imagePath := cache.LibraryImage(libraryImage.Hash, imageName)

--- a/internal/pkg/build/sources/conveyorPacker_library.go
+++ b/internal/pkg/build/sources/conveyorPacker_library.go
@@ -63,10 +63,11 @@ func (cp *LibraryConveyorPacker) Get(b *types.Bundle) (err error) {
 		return fmt.Errorf("while getting image info: %v", err)
 	}
 	if !existOk {
-		return fmt.Errorf("image does not exist in the library")
+		return fmt.Errorf("image does not exist in the library: %s", imageRef)
 	}
 	if libraryImage == nil {
-		return fmt.Errorf("failed getting image from the library")
+		// It should never get here
+		return fmt.Errorf("failed getting image from the library: %s", imageRef)
 	}
 
 	imageName := uri.GetName("library://" + imageRef)

